### ISSUE TITLE
coerce and transform not having proper types in output

### ISSUE THREAD

**Issue by teaishealthy** [RobinTail/express-zod-api#759]
Typescript reports an error when using z.coerce or .transform in output.

Additionally generating an OpenAPI spec fails at runtime when using z.coere.bigint

**Minimal reproducible example** 
```ts
import { defaultEndpointsFactory, z } from "express-zod-api";

const testEndpoint = defaultEndpointsFactory.build({
  method: "get",
  input: z.object({}),
  output: z.object({
    name: z.coerce.string(),
    id: z.coerce.bigint().
  }),
  handler: async () => {
    return { name: 1, id: "1" };  // <-  Type 'number' is not assignable to type 'string'
  },
});
```

**Comment by RobinTail** [comment]
Well-well-well...
You're assigning a number to the property having schema `z.coerce.string()`.
Typescript validates the handler's returns using `z.input< typeof z.coerce.string() >`, and it seems to be just `string`, so you're getting an error.
I'd say, that this is more likely the `zod`'s issue of the way the schemas with coercion being inferred.

But I'll check it more precisely tomorrow.
Thank you for letting me know, @teaishealthy

**Comment by RobinTail** [comment]
> when using z.coere.bigint in input
> still fails with z.coerce.bigint in input

I think it's not about `bigint`, it's about your `name` property, which is `string`, but you're assigning a number to it, @teaishealthy

**Comment by teaishealthy** [comment]
> I think it's not about `bigint`, it's about your `name` property, which is `string`, but you're assigning a number to it,

I don't think that example was directly related to the schema issue, I just found it while tinkering around. I can try to provide an example later this day.

**Comment by RobinTail** [comment]
> I don't think my provided example was directly related to the schema issue, I just found it while tinkering around. I can try to provide an example later this day.

I checked the example in details — it's fine and it really reveals the confusion with the new coercion feature in Zod.
The problem is that `coerce` feature is not reflected on the type level, it only works in runtime.

I posted a more generic issue in Zod repo with a question to Colin. Maybe they would explain should it be fixed or it's somehow intentional. 

https://github.com/colinhacks/zod/issues/1760

Anyway, thank you @teaishealthy

**Comment by RobinTail** [comment]
The issue confirmed as a bug of `zod`, @teaishealthy
https://github.com/colinhacks/zod/issues/1760#issuecomment-1365494922

**Comment by RobinTail** [comment]
I made a PR that should fix the issue in zod, but I'm still waiting for its approval.
https://github.com/colinhacks/zod/pull/1793

**Comment by RobinTail** [comment]
Bad news, @teaishealthy .
The author of `zod` confirmed the bug, but does not want to fix it, since according to the author's vision it requires a complete refactoring of `zod` to be less messy.
¯\\\_(ツ)_/¯ 
Not sure yet what I can do about it...

**Comment by teaishealthy** [comment]
That's really unfortunate :confused: although I should be able to get along without `.coere` for the time being

Anyway thank you for your efforts on this issue :+1:

**Comment by RobinTail** [comment]
Regarding `z.coerce.bigint()` — this is also third party bug:
https://github.com/colinhacks/zod/issues/1911
